### PR TITLE
[Blocks editor] Manage mixed blocks selection

### PR DESCRIPTION
**Pull request by simotae14**
### What does it do?

- Resolves the strange behavior we encountered when selecting content with mixed block types in the RTE block editor.

### Why is it needed?

Because the Blocks dropdown continuously changes its selected option between the block types in the selection.

### How to test it?

- Create a new blocks field inside a Collection.
- Create different content such as an h1, a paragraph, and a quote.
- Use the mouse to select all the created content.
- The option selected in the Dropdown will be the one related to the block type where you positioned the cursor to start the selection.

### CODE REVIEW

File: packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/tests/index.test.js [hunk 1]
```diff
@@ -259,4 +278,37 @@ describe('BlocksEditor toolbar', () => {
...
     expect(screen.getByText(title)).toBeInTheDocument();
   });
+
+  it('check if a mixed selected content show only one option selected in the dropdown', async () => {
+    setup(mixedInitialValue);
+
+    const headingsDropdown = screen.getByRole('combobox', { name: /Select a block/i });
+
+    // Set the selection to cover the entire content
+    waitFor(() => {
```
Comment (joshuaellis) [inline packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/tests/index.test.js:288]: why are you not awaiting?

Comment (simotae14) [reply packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/tests/index.test.js:288]: With await I have the console errors when I run the test asking me to wrap the instruction inside an act

Comment (remidej) [reply packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/tests/index.test.js:288]: it sounds like not awaiting is silencing the error but not really fixing it then, as far as I know waitFor is meant to be awaited to actually work

Comment (joshuaellis) [reply packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/tests/index.test.js:288]: ^^^ exactly what @remidej said, @simotae14 can you fix your tests correctly please :)

**Comment by simotae14** [comment]
Closed in favour of this new one https://github.com/strapi/strapi/pull/18181 to prevent dirty git history

**Pull request closed without merging**